### DISCUSSION TITLE
How to send routing request to apollo through code ?

### DISCUSSION THREAD

**Discussion by adityadikshit07** [guardstrikelab/carla_apollo_bridge#90] [Q&A] [answered]
Hi Team,

I am working on a project, and I want to send routing request through code instead of Dreamview UI.
Is documentation available so that I could refer to it ?
Or has anyone sent the routing request before programmatically.
It would be of help if anyone could answer.

Thanks

**Comment by WilliaJing** [comment]
hi,this question about apollo,I found a requirement similar to what you mentioned under the Apollo project, and you can refer to this issue:

https://github.com/ApolloAuto/apollo/issues/14809

I hope it can help you, of course, you can also share any experience here

**Comment by XiaoFei9704** [comment, marked as answer]
You can do like this:
`
self.routing_request_writer = self.node.create_writer('/apollo/routing_request',
                                                              routing_pb2.RoutingRequest)

routing_request = routing_pb2.RoutingRequest()
routing_request.header.timestamp_sec = now_cyber_time
routing_request.header.module_name = 'xxx'

start_apollo_waypoint = routing_request.waypoint.add()
start_apollo_waypoint.pose.x = start_location.x
start_apollo_waypoint.pose.y = start_location.y

target_apollo_waypoint = routing_request.waypoint.add()
target_apollo_waypoint.pose.x = target_location.x
target_apollo_waypoint.pose.y = target_location.y

self.routing_request_writer.write(routing_request)
`

**Reply by AOOOOOA** [reply]
> You can do like this: ` self.routing_request_writer = self.node.create_writer('/apollo/routing_request', routing_pb2.RoutingRequest)
> 
> routing_request = routing_pb2.RoutingRequest() routing_request.header.timestamp_sec = now_cyber_time routing_request.header.module_name = 'xxx'
> 
> start_apollo_waypoint = routing_request.waypoint.add() start_apollo_waypoint.pose.x = start_location.x start_apollo_waypoint.pose.y = start_location.y
> 
> target_apollo_waypoint = routing_request.waypoint.add() target_apollo_waypoint.pose.x = target_location.x target_apollo_waypoint.pose.y = target_location.y
> 
> self.routing_request_writer.write(routing_request) `

Hi, where should we place this block of code to assign the destination? Thank you!
